BUILD.bazel

```diff
@@ -53,24 +54,95 @@ cc_library(
         "src/pcre2_valid_utf.c",
         "src/pcre2_xclass.c",
         ":pcre2_chartables_c",
-    ],
-    hdrs = glob(["src/*.h"]) + [
+        "src/pcre2_compile.h",
+        "src/pcre2_internal.h",
+        "src/pcre2_intmodedep.h",
+        "src/pcre2_ucp.h",
+        "src/pcre2_util.h",
         ":config_h_generic",
+    ],
+    textual_hdrs = [
+        "src/pcre2_jit_match.c",
+        "src/pcre2_jit_misc.c",
+        "src/pcre2_ucptables.c",
+    ],
+    hdrs = [
         ":pcre2_h_generic",
     ],
     defines = [
         "HAVE_CONFIG_H",
+        "HAVE_MEMMOVE",
         "PCRE2_CODE_UNIT_WIDTH=8",
         "PCRE2_STATIC",
+        "SUPPORT_UNICODE",
     ],
     includes = ["src"],
     strip_include_prefix = "src",
     visibility = ["//visibility:public"],
 )
 
+cc_library(
+    name = "pcre2-posix",
+    srcs = [
+        "src/pcre2posix.c",
+        ":config_h_generic",
+    ],
+    hdrs = [
+        "src/pcre2posix.h",
+    ],
+    defines = [
+        "HAVE_CONFIG_H",
+        "HAVE_MEMMOVE",
+        "PCRE2_CODE_UNIT_WIDTH=8",
+        "PCRE2_STATIC",
+        "SUPPORT_UNICODE",
+    ],
+    includes = ["src"],
+    strip_include_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [":pcre2"]
+)
+
+# Totally weird issue in Bazel. It won't let you #include any files unless they
+# are declared to the build system. OK, fair enough. But - for a cc_binary it
+# uses the file extension to determine whether it's a header or a compilation
+# unit. But... we have several .c files which are #included, rather than treated
+# as a compilation unit.
+#
+# For cc_library() above, we can overcome this with textual_hdrs. But that
+# doesn't work for cc_binary(). Here's our workaround.
+#
+# https://github.com/bazelbuild/bazel/issues/680
+cc_library(
+    name = "pcre2test_dotc_headers",
+    hdrs = [
+        "src/pcre2_chkdint.c",
+        "src/pcre2_printint.c",
+        "src/pcre2_tables.c",
+        "src/pcre2_ucd.c",
+        "src/pcre2_valid_utf.c",
+    ],
+    strip_include_prefix = "src",
+    visibility = ["//visibility:private"],
+)
+
 cc_binary(
```

Comment:
typically a test would be done through a `cc_test` and then ran with `bazel test //...`.

Reply:
I have granted you access to the repo.

Please, go ahead and make it work, if you can. I spent quite a while yesterday trying to do that, and totally gave up.

What we need to do is really, really simple, and would be a one-liner in any build system other than Bazel. We just need to run `pcre2test=PATH_TO_OUTPUT ./RunTest` in the source directory. The test passes if the RunTest script has a zero exit status. The RunTest script can be passed the path to the `pcre2test` binary, and it looks in the current directory for the `testdata/` folder. However, this seems super-hard to do in Bazel.

Reply:
Thanks for letting me know how it works! A `sh_test` would be correct in that case. Bazel also needs to know about all a tests inputs when it runs and it does that throught the `data` attribute. I just globbed all the data in the `testdata` directory and it seems to work. Usually you would do it through bazel runfiles, but with how pcre2's file structure it _should_ work as is.

Reply:
Thank you. I tried for a bit to get it work on Windows too. We have a `RunTest.bat` file for Windows (so it shouldn't need `bash` or MSYS at all). But I couldn't work out how to execute a batch file in bazel! There's a `native_test` helper in Skylib, which will run an exe as a test, but that can't run a batch script.

I find this all very weird, that a supposedly mature build system won't let you define your tests as a script, or give you good tools for just "execute this".

I will merge this PR, if you are happy with the lack of testing on Windows.

Reply:
the `native_test` can actually run a `.bat` file (you just have to set the `out` attribute). I gave it a try, but it failed to find the `pcre2test.exe`. I'd have to look into it more to see how to properly find it.

If I get the time I could do a Windows follow-up, yeah. This is already a great improvement.

Reply:
the `sh_test` is _supposed_ to work for windows batch files as well, but I ran into this issue https://github.com/bazelbuild/bazel/issues/20345

Reply:
OK. These Bazel issues are taking too much time for me to fix. I will merge this, without Windows support, and if you wish to add it later you are very welcome to open a PR for that.

Reply:
Thank you very much for your help in this PR!